### PR TITLE
DOC: Summarize printing changes in release notes

### LINKED ISSUE #10058: Add a section to the release notes summarizing all the formatting changes
A lot of stuff is in there already, but it would be good to group it all together.

Would also be good to explain that `np.set_printoptions(legacy='1.13')` is primarily for transitioning doctests from pre- to post- 1.14

### PR DESCRIPTION
**Pull request by ahaldane**
Fixes #10058

Suggestions and modification welcome.

### CODE REVIEW

File: doc/release/1.14.0-notes.rst [hunk 1]
```diff
@@ -193,19 +193,42 @@ raising a ``TypeError``.
...
+   in the sign position. See the new ``sign`` option to ``np.set_printoptions``.
+ * Floating-point arrays use a new algorithm for their decimal representations,
+   allowing floats to be represented "uniquely".  This means ``float16`` arrays
+   will print with fewer digits. ``float64`` arrays should be unaffected. See
+   the new ``floatmode`` option to ``np.set_printoptions``.
+ * Floating-point scalars now print with exactly the number of digits needed to
+   be uniquely specified, so ``float16`` will be shorter and ``float128`` 
+   longer. The ``str`` of ``float64`` scalars is no longer truncated in python2.
+ * ``NaT`` values in datetime arrays are now properly aligned
+ * ``MaskedArray`` arrays have a new single-line output mode for short arrays
```
Comment (eric-wieser) [inline doc/release/1.14.0-notes.rst]: This isn't true - they always wrap to another line. What changed is they:
* Show the dtype
* Include commas
* Correctly align wrapped arrays

File: doc/release/1.14.0-notes.rst [hunk 2]
```diff
@@ -193,19 +193,42 @@ raising a ``TypeError``.
...
+   and a longer left-justified output mode for long arrays.
+ * Arrays and scalars of ``np.void`` datatype now print using hex notation.
+ * 0d arrays no longer have their own idiosyncratic implementations of ``str``
+   and ``repr``. The ``style`` argument to ``np.array2string`` is deprecated.
+ * Size-0 arrays with multiple dimensions like ``np.zeros((2,0,1))`` are 
+   represented using ``empty`` instead of ``array``.
```
Comment (eric-wieser) [inline doc/release/1.14.0-notes.rst]: Pending on #10119, anyway...

File: doc/release/1.14.0-notes.rst [hunk 3]
```diff
@@ -193,19 +193,42 @@ raising a ``TypeError``.
...
+ * Arrays of ``object`` datatype containing python lists now clearly 
+   represents the list using the ``list()`` function.
```
Comment (eric-wieser) [inline doc/release/1.14.0-notes.rst]: This isn't new - it was part of 1.13

File: doc/release/1.14.0-notes.rst [hunk 4]
```diff
@@ -193,19 +193,42 @@ raising a ``TypeError``.
...
+ * Arrays of ``bool`` datatype no longer print the datatype in the repr.
+ * The "dtype" part of ndarray reprs will now be printed on the next line
+   if there isn't space on the last line of array output.
+ * User-defined types now need to implement ``__str__`` and ``__repr__``
```
Comment (eric-wieser) [inline doc/release/1.14.0-notes.rst]: `dtypes` or "subclasses of `np.generic`" would be clearer

**Comment by eric-wieser** [comment]
Don't forget the `...,` in #9815

**Comment by ahaldane** [comment]
updated

I removed the line about `empty` for now, we need to add it back if we go through with #10119

**Comment by eric-wieser** [comment]
Looks good. Are there any other sections of the release notes that can be removed as a result of this list?

**Comment by eric-wieser** [comment]
Maybe needs a note about #10129 now too

**Comment by mhvk** [comment]
This reads very well!

**Review by eric-wieser — APPROVED** [review]

**Comment by charris** [comment]
Thanks Allan. I noticed some spelling nits, but will run a spell checker over this before branching 1.14 and maybe make some other small changes.

**Pull request merged**